crates/frontend/src/components/default_config_form.rs

```diff
@@ -226,49 +213,84 @@ where
                             .into_view()
                     }
                     "BOOLEAN" => {
-                        let value = config_value.get();
-                        if value.is_empty() {
-                            set_config_value
-                                .set(value.parse::<bool>().unwrap_or(false).to_string());
+                        let value = config_value_rs.get();
+                        if value.is_null() {
+                            config_value_ws.set(Value::Bool(false));
                         }
                         view! {
                             <BooleanToggle
-                                value={value.parse::<bool>().unwrap_or(false)}
-                                on_change=Callback::new(move |flag: bool| {
-                                    set_config_value.set(flag.to_string());
-                                })
+                                value=value.as_bool().unwrap_or(false)
```

Comment:
@Datron , the pattern matching can be removed and in place the Input component can be used.
```rust
let schema_type = SchemaType::try_from(schema);
let enum_variants = EnumVariants::try_from(schema);
let input_type = InputType::from((schema_type, variants));
```
These snippets can be used to determine the required type and enum options, and then can be directly passed to the `Input` component.

```rust
<Input
    id=id
    class=""
    r#type=input_type.unwrap()
    value=value
    schema_type=schema_type.unwrap()
    on_change=move |value| {}
/>
```